### ISSUE TITLE
A transaction can stop block prodution

### ISSUE THREAD

**Issue by Smittyvb** [nimiq-network/core#434]
### General information

- Library version(s): all
- Browser version(s): all
- Devices/Simulators/Machine affected: all
- Reproducible in the testnet? (Yes/No): Yes
- Related issues: none

#### Expected behavior

> This transaction does not stop block production.

#### Actual behavior

> This transaction stops block production.

#### Steps to reproduce

> Try to mine

#### Crash log? Screenshots? Videos? Sample project?

![image](https://user-images.githubusercontent.com/10530973/44622102-254ec680-a880-11e8-9230-d177405f497a.png)
Tx: https://nimiq.watch/#4e07c029eb64b2c6368e5e2d43cb8174869e5ae19f82511ebce45feb8596dadd

**Comment by Smittyvb** [comment]
Users are getting this error:
```error
Error: Invalid block body: Account was invalidly pruned
    at BasePoolMiner._getNextHeader
```

**Comment by sisou** [comment]
The above transaction is a HTLC creation transaction and was likely not directly responsible for the mining problem.

A quick fix was released in v1.3.1 here: https://github.com/nimiq-network/core/releases/tag/v1.3.1
